src/fu-main.c

```diff
@@ -43,8 +44,23 @@ typedef struct {
 	PolkitAuthority		*authority;
 	guint			 owner_id;
 	FuEngine		*engine;
+	gboolean		 update_in_progress;
+	gboolean		 pending_sigterm;
 } FuMainPrivate;
 
+static gboolean
+fu_main_sigterm_cb (gpointer user_data)
+{
+	FuMainPrivate *priv = (FuMainPrivate *) user_data;
+	if (!priv->update_in_progress) {
```

Comment:
Yes, I think if we end up connecting this to other actions beyond update yes we should move it to the engine in a safe way.  For now I think this way is safer since it lets you guarantee that the engine is fully done with the update process (success or not).

I'll let CI confirm it looks good and then merge.